### PR TITLE
[Sprite] [TG Port] Things that go on your feet

### PR DESCRIPTION
**Pull request by Aurorablade**
Ports https://github.com/tgstation/-tg-station/pull/6528 with my addition of a crappy version of fish nets

Issues:
I am iffy on the savefile stuff as i do not have a local sql setup.

Other changes:
On sec records, i changed it to crime, from disability..

Screenshot on races that can wear underwear:

space station 13 2015-08-21 173725

@tigercat2000 I await for you to tell me what i did wrong and smack me with a stick.

### CODE REVIEW

**Comment by TheDZD** [comment]
Looks like you stole @tigercat2000's thunder.

**Comment by Fox-McCloud** [comment]
@Aurorablade 

Just to point somethin' out; since we use SQL for everything, changes to the savefile won't be necessary.

**Comment by tigercat2000** [comment]
Uh, yeah, this needs SQL edits.

**Comment by Earthdivine** [comment]
>Other changes:
On sec records, i changed it to crime, from disability

I don't know if this should be on the same PR as this?

**Comment by Aurorablade** [comment]
@tigercat2000 well i can close this and let you handle it if you want. sorry :( i will try to get a local SQL up later maybe.
@Fox-McCloud figured cuase it wasn't ticked but i wasn't sure.

@Earthdivine Good point i will split it out.

**Comment by tigercat2000** [comment]
@Aurorablade No need for that, I can help with any SQL changes, you are the one who ported it.

**Comment by Aurorablade** [comment]
@Fox-McCloud should i revert the changes to the savefile.dm also?

....And anyone want any other color variations? Can't make a new sprite but i can edit THE HELL out of them.

**Comment by tigercat2000** [comment]
@Aurorablade Eh. The savefile is entirely broken as is, doesn't really matter.

**Comment by Aurorablade** [comment]
okay @tigercat2000 iwent though where you did the underwear refactor and the sql changes what i am unsure of is in prefrences_mysql.dm with the querys, Been a while since i have touched sql but i see they are numbered now my brain is telling me to either renumber to make room for socks or add socks at the last query entry with a new number..but my brain is also telling me...this could lead to !!CodeGriff!! from the code.

**Comment by tigercat2000** [comment]
@Aurorablade It would be better to tack it on as a new entry, than to budge everything. That would require the server database to be significantly updated, and cause a mess. Much smaller mess just having underwear, undershirt, and socks being seperated.

**Comment by Aurorablade** [comment]
@tigercat2000 Alright will do..I am also gonna have to take the HAS_UNDERWEAR check off..or make it so slimepeople can wear them also..or Dreadweaver will murder me...just need to use the species allowed list.

**Comment by tigercat2000** [comment]
@Aurorablade The HAS_UNDERWEAR check should probably be scrapped in general. It doesn't even prevent people from selecting underwear.

**Comment by TheDZD** [comment]
If you do remove the HAS_UNDERWEAR flag, you will want to make sure that having underwear/shirts/socks on certain species (Vox and IPC come to mind) doesn't cause horrendous clipping.

**Comment by Aurorablade** [comment]
I am just going to remove the HAS_UNDERWEAR for socks. SOCKS FOR EVERYONE
However I am iffy on kiden haveing socks.....and i will have to sprite edit a few voxxy socks. THEN i need to make sure only voxxy can wear voxxy socks. I am unsure the best means to do this yet.

as setting just species allowed doesn't work. it looks like i am going to have to restructure it like how the hair is for some of the races (like quills, etc etc)

space station 13 2015-08-22 225533

**Comment by Keilden** [comment]
IPC socks! 

:+1: :100:

**Comment by TheDZD** [comment]
@Aurorablade You might want to see how species-specific hair is handled, as you could probably do something similar to that.

**Comment by Aurorablade** [comment]
think i figured it out. Have to do more after Work though.

**Comment by Aurorablade** [comment]
@tigercat2000 To explain the pathing, as my next sync is gonna move the code comment. I wad doing it to do race restriction but now that i think about it...I wasn't checking for species allowed before in the sock code in one file. Now i am. So i am gonna try to go back to relative paths and check it out. Assuming github stops being DDoSed

**Comment by KasparoVy** [comment]
HELL YEAH BRO
FOXES WITH BOXES OF VOX SOCKSES

:+1: :x: :100:

**Comment by Aurorablade** [comment]
Okay any other color variations i need to add/code chages? Also one think i did note. The sprite preiview on the char setup IS NOT updateing. and this is currently on the Live Server also so its no this PR that did it..I am however unsure how to fix.

**Comment by Alex6511** [comment]
@Aurorablade any bug on the live server needs an issue report not a comment in a (probably) unrelated PR.

**Comment by Aurorablade** [comment]
@Alex6511 Well i figured i could try to fix it. but i do see your point. Anyway, Anyone want to suggest color variations or other things i can do cuase it might be done unless @tigercat2000 throws a code suggestion/change/tweak at me.

File: code/modules/mob/new_player/sprite_accessories.dm [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/*
+/paradise_schema.sql/*
```
Comment (tigercat2000) [inline code/modules/mob/new_player/sprite_accessories.dm]: this isn't actually commented out

**Comment by tigercat2000** [comment]
Other than my last note, all seems just fine to me. You might want to make a screenshot of all the socks, but that's up to you.

File: code/modules/mob/new_player/sprite_accessories.dm (hunk 1 above)
Comment (Aurorablade) [inline code/modules/mob/new_player/sprite_accessories.dm]: ...i...don't event know why that is like that......unless i accidently delete an *

Comment (Aurorablade) [inline code/modules/mob/new_player/sprite_accessories.dm]: no i see what it is...i accidently pasted it in while i was doing a file search....DOH

**Comment by Aurorablade** [comment]
space station 13 2015-08-29 192749
a good chunk of the styles, to recap them:
Short,Normal,Thigh/Knee high:
White
Black
Thin (has a bit of a transparency)
Striped
Rainbow

Unique:
Pantyhose
Black Fish nets

Vox Normal and Thigh High(NOT THE BEST SPRITES EVER):
Black
White
Thin
Striped
Rainbow
FishNets

also @tigercat2000 the Dresser is not respecting Gender/race species (i put a blad greytide male in kinky femlae underwear) and i was unsure how to fix..so expect sock and underwear hilarity....

**Comment by tigercat2000** [comment]
@Aurorablade I wouldn't call that a problem.

**Comment by Aurorablade** [comment]
well the underwera thing is actually hilarious...as it will relate to socks..it might still work...and be equally as hilarious but still.

**Comment by Spacemanspark** [comment]
Vox has many socks, ya ya?
Is wear all the times!

**Comment by Iamgoofball** [comment]
fetish addition

:-1:

**Comment by Spacemanspark** [comment]
what.

**Comment by Iamgoofball** [comment]
please explain the reasoning behind adding socks to Space Station 13

**Comment by Spacemanspark** [comment]
I don't know, it's just one of those things that makes the game amusing.
/tg/ has it, does it not?

**Comment by Fox-McCloud** [comment]
Customization and expression of individuality? I opposed this at one point (for some clothing anyway), and was *dead wrong* about it. Yes, it does generate some minor issues, but the ability for players to be able to customize their own character, to their liking, is a positive addition to the game. When you only have 2 dimensions and 32x32 pixels to express who/what your character is, every little bit/distinction helps.

I don't see how customization, especially socks, is a bad thing.

**Comment by Iamgoofball** [comment]
i'm pretty sure the only reason socks were added at /tg/ was someone shared the opinion of Quentin Tarantino,  but whatever

**Comment by dreadweaver** [comment]
CANNOT COME FAST ENOUGH

**Comment by Aurorablade** [comment]
this is probaly ready for a last review before merge.

File: code/modules/mob/new_player/sprite_accessories.dm [hunk 2]
```diff
@@ -1260,4 +1260,193 @@
...
+	icon_state = "pantyhose"
+	gender = FEMALE
+	species_allowed = list("Human","Unathi","Diona","Grey","Machine","Tajaran","Vulpakanin","Slime People","Skellington")
+
+
+/datum/sprite_accessory/socks/black_fishnet
+	name = "Black Fishnet"
+	icon_state = "black_fishnet"
+	gender = NEUTER
+	species_allowed = list("Human","Unathi","Diona","Grey","Machine","Tajaran","Vulpakanin","Slime People","Skellington")
```
Comment (TheDZD) [inline code/modules/mob/new_player/sprite_accessories.dm]: Skellingtons in fishnets? *shudder*

**Comment by TheDZD** [comment]
:+1: Looks good to be merged once SQL changes can be arranged for near the time of a server update.

**Comment by Aurorablade** [comment]
yup also this is how command staff will look http://3.bp.blogspot.com/-buVrLP0pjPc/UnAV2kr-dmI/AAAAAAAAP8Y/fxsQXt9LBAE/s1600/Rocky-sweet_transvestite.jpg

**Comment by Fox-McCloud** [comment]
Whyyyyy @Aurorablade

**Comment by Markolie** [comment]
I'll contact Earth and get the SQL changes done so it can be merged.

**Comment by FalseIncarnate** [comment]
:-1: because I just realized this means socks + sandals.

OH GOD NO. MAKE SANDALS HIDE SOCKS!

**Comment by tigercat2000** [comment]
*scream

**Comment by TheDZD** [comment]
OH GOD NO NOT SANDALS WITH SOCKS.

**Comment by Fox-McCloud** [comment]
*scream

**Pull request merged**